### ISSUE TITLE
Library crashes the program when calling `cpr::Session::Post` multiple times with a big body

### ISSUE THREAD

**Issue by egor-bychin-sd** [libcpr/cpr#970]
### Description

I've made a small program that reproduces the crash:
```C++
#include <cpr/session.h>

int main()
{
    std::ifstream ifs{ R"(C:\dev\request_body.txt)" };
    std::stringstream ss;
    ss << ifs.rdbuf();
    cpr::Body body{ ss.str() };
    cpr::Url url{ "https://httpbin.org/post" };
    cpr::Session session;
    session.SetOption( url );
    session.SetOption( body );
    session.Post();
    session.Post(); // The app crashes on this line due to a reading access violation
}
```

[request_body.txt](https://github.com/libcpr/cpr/files/13023351/request_body.txt) is a 3.81 megabyte file, that consists only of letter `a` repeated 4 million times.

What's interesting is that this bug only reproduces if the body is big. I tried to reproduce it with a file consisting only of 1700 `a`s, but had no luck.

Also, if we call `session.SetOption(body)` again before the second invocation of `session.Post`, then everything works as expected:
```C++
#include <cpr/session.h>

int main()
{
    std::ifstream ifs{ R"(C:\dev\request_body.txt)" };
    std::stringstream ss;
    ss << ifs.rdbuf();
    cpr::Body body{ ss.str() };
    cpr::Url url{ "https://httpbin.org/post" };
    cpr::Session session;

    session.SetOption( url );
    session.SetOption( body );
    session.Post();

    session.SetOption( body );
    session.Post(); // No crash here
}
```

### Example/How to Reproduce

1. Create a `cpr::Session`
2. Set a huge body and a URL
3. Perform the POST request twice
4. See the app crash

### Where did you get it from?

vcpkg

### Additional Context/Your Environment

- OS: Windows 11 Pro 22H2
- Version: Here's the output of `./vcpkg list` command
```
cpr:x64-windows                                   1.10.2+3            c++ requests is a simple wrapper around libcurl ...
cpr[ssl]:x64-windows                                                  enable ssl support
curl:x64-windows                                  8.4.0               a library for transferring data with urls
curl[non-http]:x64-windows                                            enables protocols beyond http/https/http2
curl[schannel]:x64-windows                                            ssl support (secure channel)
curl[ssl]:x64-windows                                                 default ssl backend
curl[sspi]:x64-windows                                                sspi support
```

**Comment by COM8** [comment]
@egor-bychin-sd Thanks for reporting and the good issue!
I will look into it. Sadly I won't have time for it until Saturday or Sunday.

An workaround would be recreating the `cpr::Session` object and then sending again.

Is there any special importance (time) to this issue from your side?

**Comment by egor-bychin-sd** [comment]
@COM8 no, I already implemented this workaround in the software I am currently working on

**Comment by COM8** [comment]
Fixed in #973. Could you please verify?

**Comment by egor-bychin-sd** [comment]
I'm not sure when I'll have time to verify this, because it'd require me to build the whole thing myself. I'll try to do that in like two days, anyway. For now I'll look at the patch you've made.
